### ISSUE TITLE
How can I increase 75000 hard coded limit?

### ISSUE THREAD

**Issue by MonsterMMORPG** [linvi/tweetinvi#737]
for example in this file

https://github.com/linvi/tweetinvi/blob/905b1cbc8674b718c3e64cbeb2fc96cc6cb4efbd/Tweetinvi.Controllers/Friendship/FriendshipController.cs

 public IEnumerable<IUser> GetUsersYouRequestedToFollow(int maximumUsersToRetrieve = **75000**)

**Comment by linvi** [comment]
This is not a limit, this is a default value. Also if I remember properly it should be 7500 and not 75000.

**Comment by MonsterMMORPG** [comment]
@linvi my account has about 77.000 followers and i am able to fetch maximum 75.000 even though i provide higher value

how do you explain?

it fetches exactly 75.000

**Comment by linvi** [comment]
Hi. 

I guess that the problem you are experiencing is due to RateLimits even though I can see in your code that you properly set the `RateLimitTrackerMode` .

I looked at the code and everything is supposed to work.

I never had an account with more than 75k users to test my code against.
If you can share the credentials temporarily for me to debug the issue I will be happy to do so.

You can contact me privately on gitter : https://gitter.im/linvi/

Cheers,
Linvi

**Comment by MonsterMMORPG** [comment]
Probably at some point parameter is not passed so default 75.000 is used

After i have increased entire solution all 75.000 to 1.000.000 and recompiled and replaced new DLLs, now it works as supposed to be

**Comment by linvi** [comment]
Thanks, this will be taken care of as part of release 4.0 or 5.0.

**Comment by JoshKeegan** [comment]
Hi @MonsterMMORPG,
I've opened a PR (#764) with a fix for this. Can you try it out and see if your issue is fixed?
Here's a build:
[netstandard1.4.zip](https://github.com/linvi/tweetinvi/files/2499416/netstandard1.4.zip)

**Comment by MonsterMMORPG** [comment]
hi it wasnt after all 75000 but the delay issue

so after 75000 (which is - 15*5000) your quote fills

and the application has to wait

i cant remember now where i changed but i have added extra wait to the consquent actions and that problem gone away

so the problem was definitely about waiting

since it doesnt wait properly, it was causing no action quote query and un-caught error

**Comment by JoshKeegan** [comment]
I wrote the following as a quick repro:
```cs
ExceptionHandler.SwallowWebExceptions = false;
RateLimit.RateLimitTrackerMode = RateLimitTrackerMode.TrackAndAwait;

IEnumerable<long> followerIds = User.GetFollowerIds("justinbeiber", 100000);
Console.WriteLine("Fetched {0} follower IDs", followerIds.Count());
```
and I got an exception! 

It looks like the problem was that Tweetinvi was waiting the exact number of milliseconds that Twitter told us to wait, but due to timing differences between our machine, the server that served the last response (telling us how long to wait) and the server we send the request to (after waiting), you might end up sending your request marginally too early for Twitter to accept it. To compensate, I've increased the wait time by an extra 5s (I first tried 1s, but still got an error), which should be sufficient to allow for any reasonable timing discrepancy without slowing things down too much :).  
You can also override the 5s additional wait time with your own preferred value:
```csharp
TweetinviConfig.CurrentThreadSettings.RateLimitWaitFudgeMs = 10000;
```

The PR has been updated, and here's another build:
[netstandard1.4.zip](https://github.com/linvi/tweetinvi/files/2507841/netstandard1.4.zip)
Can you see if this works for you?

**Comment by MonsterMMORPG** [comment]
@JoshKeegan yep this was the problem i have discovered :)

i have added if i remember correctly 5000 ms dealy and never had problem since then (it has been months)

i can try later your update thank you

**Comment by linvi** [comment]
Hello there,

This has now been merged into 5.0.
Thanks for all your work and help.

Linvi
